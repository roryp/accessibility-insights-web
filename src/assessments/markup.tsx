// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

export function Tag(props: { tagName: string; isBold?: boolean }): JSX.Element {
    const isBold = props.hasOwnProperty('isBold') ? props.isBold : true;
    return isBold ? (
        <CodeTerm>&lt;{props.tagName}&gt;</CodeTerm>
    ) : (
        <Code>&lt;{props.tagName}&gt;</Code>
    );
}

export function Emphasis(props: { children: React.ReactNode }): JSX.Element {
    return <em>{props.children}</em>;
}

export function Term(props: { children: React.ReactNode }): JSX.Element {
    return <strong>{props.children}</strong>;
}

export function Code(props: { children: React.ReactNode }): JSX.Element {
    return <span className="insights-code">{props.children}</span>;
}

export function CodeBlock(props: { children: React.ReactNode }): JSX.Element {
    return <div className="insights-code">{props.children}</div>;
}

export const GreaterThanOrEqualTo = () => <span>&#8805;</span>;

export const NonBreakingSpace = () => <span>&nbsp;</span>;

export function CodeTerm(props: { children: React.ReactNode }): JSX.Element {
    return (
        <Term>
            <Code>{props.children}</Code>
        </Term>
    );
}
