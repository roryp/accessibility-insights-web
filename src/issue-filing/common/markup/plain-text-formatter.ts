// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { MarkupFormatter } from './markup-formatter';

const createFormatter = (): MarkupFormatter => {
    const snippet = (text: string): string => text.replace(/\s+/g, ' ').trim();

    const link = (href: string, text?: string): string => {
        if (text) {
            return `${text} - ${href}`;
        }

        return href;
    };

    const sectionHeader = (text: string): string => text;

    const howToFixSection = (failureSummary: string): string => `\n${failureSummary}`;

    const sectionHeaderSeparator = () => ': ';

    const footerSeparator = () => '====\n\n';

    const newLine = () => '\n\n';

    return {
        snippet,
        link,
        sectionHeader,
        howToFixSection,
        sectionHeaderSeparator,
        footerSeparator,
        sectionSeparator: newLine,
    };
};

export const PlainTextFormatter = createFormatter();
