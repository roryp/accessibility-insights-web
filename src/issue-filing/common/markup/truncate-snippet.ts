// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { truncate } from 'lodash';
export const maxSnippetLength = 256;

export const truncateSnippet = (text: string) => truncate(text, { length: maxSnippetLength });
