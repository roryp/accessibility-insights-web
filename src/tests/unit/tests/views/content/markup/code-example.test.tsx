// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';

import { CodeExample } from 'views/content/markup/code-example';

describe('<CodeExample>', () => {
    it('renders with title', () => {
        const wrapper = shallow(<CodeExample title="title">code</CodeExample>);
        expect(wrapper.find('.code-example-title').getElement()).toEqual(
            <div className="code-example-title">
                <h4>title</h4>
            </div>,
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders with no title', () => {
        const wrapper = shallow(<CodeExample>code</CodeExample>);
        expect(wrapper.find('.code-example-title').exists()).toEqual(false);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders with no highlighted region', () => {
        const wrapper = shallow(<CodeExample>No highlight</CodeExample>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders with one highlighted region', () => {
        const wrapper = shallow(<CodeExample>One [single] highlight</CodeExample>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders with empty highlighted region', () => {
        const wrapper = shallow(<CodeExample>Empty [] highlight</CodeExample>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders with unterminated highlighted region', () => {
        const wrapper = shallow(<CodeExample>One [unterminated highlight</CodeExample>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders with many highlighted regions', () => {
        const wrapper = shallow(
            <CodeExample>With [quite] a [number] of [highlights].</CodeExample>,
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders with a single line break', () => {
        const wrapper = shallow(<CodeExample>{`Line 1\nLine 2`}</CodeExample>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders with a line breaks and a highlighted line', () => {
        const wrapper = shallow(<CodeExample>{`Line 1\n[Line 2]\nLine 3`}</CodeExample>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders with a highlight that spans multiple lines', () => {
        const wrapper = shallow(<CodeExample>{`Line 1\n[Line 2\nLine 3]\nLine 4`}</CodeExample>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders with a highlight that breaks in the middle of multiple lines', () => {
        const wrapper = shallow(
            <CodeExample>{`Line 1\nLine 2 [HIGHLIGHT\nHERE]Line 3\nLine 4`}</CodeExample>,
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('br has the correct key', () => {
        const wrapper = shallow(<CodeExample>{`Line 1\nLine 2`}</CodeExample>);
        expect(wrapper.find('br').at(0).key()).toEqual(`line-breaker-1`);
    });
});
