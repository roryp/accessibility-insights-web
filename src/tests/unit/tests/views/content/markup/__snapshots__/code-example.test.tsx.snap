// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeExample> renders with a highlight that breaks in the middle of multiple lines 1`] = `
<div
  className="code-example"
>
  <div
    className="code-example-code"
  >
    <CodeBlock>
      Line 1
      <br />
      Line 2 
      <span
        className="highlight"
      >
        HIGHLIGHT
        <br />
        HERE
      </span>
      Line 3
      <br />
      Line 4
    </CodeBlock>
  </div>
</div>
`;

exports[`<CodeExample> renders with a highlight that spans multiple lines 1`] = `
<div
  className="code-example"
>
  <div
    className="code-example-code"
  >
    <CodeBlock>
      Line 1
      <br />
      
      <span
        className="highlight"
      >
        Line 2
        <br />
        Line 3
      </span>
      
      <br />
      Line 4
    </CodeBlock>
  </div>
</div>
`;

exports[`<CodeExample> renders with a line breaks and a highlighted line 1`] = `
<div
  className="code-example"
>
  <div
    className="code-example-code"
  >
    <CodeBlock>
      Line 1
      <br />
      
      <span
        className="highlight"
      >
        Line 2
      </span>
      
      <br />
      Line 3
    </CodeBlock>
  </div>
</div>
`;

exports[`<CodeExample> renders with a single line break 1`] = `
<div
  className="code-example"
>
  <div
    className="code-example-code"
  >
    <CodeBlock>
      Line 1
      <br />
      Line 2
    </CodeBlock>
  </div>
</div>
`;

exports[`<CodeExample> renders with empty highlighted region 1`] = `
<div
  className="code-example"
>
  <div
    className="code-example-code"
  >
    <CodeBlock>
      Empty 
      <span
        className="highlight"
      >
        
      </span>
       highlight
    </CodeBlock>
  </div>
</div>
`;

exports[`<CodeExample> renders with many highlighted regions 1`] = `
<div
  className="code-example"
>
  <div
    className="code-example-code"
  >
    <CodeBlock>
      With 
      <span
        className="highlight"
      >
        quite
      </span>
       a 
      <span
        className="highlight"
      >
        number
      </span>
       of 
      <span
        className="highlight"
      >
        highlights
      </span>
      .
    </CodeBlock>
  </div>
</div>
`;

exports[`<CodeExample> renders with no highlighted region 1`] = `
<div
  className="code-example"
>
  <div
    className="code-example-code"
  >
    <CodeBlock>
      No highlight
    </CodeBlock>
  </div>
</div>
`;

exports[`<CodeExample> renders with no title 1`] = `
<div
  className="code-example"
>
  <div
    className="code-example-code"
  >
    <CodeBlock>
      code
    </CodeBlock>
  </div>
</div>
`;

exports[`<CodeExample> renders with one highlighted region 1`] = `
<div
  className="code-example"
>
  <div
    className="code-example-code"
  >
    <CodeBlock>
      One 
      <span
        className="highlight"
      >
        single
      </span>
       highlight
    </CodeBlock>
  </div>
</div>
`;

exports[`<CodeExample> renders with title 1`] = `
<div
  className="code-example"
>
  <div
    className="code-example-title"
  >
    <h4>
      title
    </h4>
  </div>
  <div
    className="code-example-code"
  >
    <CodeBlock>
      code
    </CodeBlock>
  </div>
</div>
`;

exports[`<CodeExample> renders with unterminated highlighted region 1`] = `
<div
  className="code-example"
>
  <div
    className="code-example-code"
  >
    <CodeBlock>
      One 
      <span
        className="highlight"
      >
        unterminated highlight
      </span>
    </CodeBlock>
  </div>
</div>
`;
